### ISSUE TITLE
Post Model Timezone test issue

### ISSUE THREAD

**Issue by acburdine** [TryGhost/Ghost#6975]
### Issue Summary

When running server tests locally (using `grunt validate`) I continue to get an error that appears to be related to timezones:

The mocha test that fails: `Post Model Single author posts findOne can findOne, returning a dated permalink:`

The error message itself:

```
AssertionError: expected '/2015/01/01/html-ipsum/' to be '/2014/12/31/html-ipsum/'
+ expected - actual

-/2015/01/01/html-ipsum/
+/2014/12/31/html-ipsum/
```

### Steps to Reproduce

1. Clone 👻 
2. `npm install`
3. `grunt validate`
4. get error

It also occurs when running `grunt test-integration`. I tried deleting the `ghost-test.db` file in `content/data`, but that didn't affect anything.

### Technical details:

* Ghost Version: 0.9.0-beta.1 (also latest [master](https://github.com/TryGhost/Ghost/commit/451c29d06a65d3904470b6b30f2fb21ad5c162c2))
* Node Version: v0.12.2
* Browser/OS: Mac OSX El Capitan (10.11.5)
* Database: SQLite

**Comment by acburdine** [comment]
Screenshot of error:

![screen shot 2016-06-11 at 2 08 01 pm](https://cloud.githubusercontent.com/assets/5167581/15987214/fb26d11c-2fdd-11e6-9533-391040fe1c1b.png)

**Comment by kirrg001** [comment]
I am not able to reproduce this error.
Can you ensure it also fails when you run it as single test `grunt test:integration/model/model_posts_spec.js`?

**Comment by ErisDS** [comment]
Sorry @kirrg001 there was also a conversation about this in the [#dev channel](https://ghost.slack.com/archives/dev/p1465674696000674), should have come back and posted an update.

This happens if you change your local timezone to something a bit behind UTC  - e.g. something American.

I think it is related to this issue you raised: https://github.com/TryGhost/Ghost/issues/6930. All the fixtures in the [data generator](https://github.com/TryGhost/Ghost/blob/master/core/test/utils/fixtures/data-generator.js#L13) use `new Date()` and shouldn't?

**Comment by kirrg001** [comment]
Thanks for info @ErisDS 

Because our test env instantiate the post dates with `new Date(....)`, the dates are correctly stored (because for example we tell mysql to store UTC dates, see `data/db/connection`), but the test itself gets a local JS Date back from the data generator, creates a permalink from this local date and tries to compare it with a permalink which is created based on your blog TZ. So there are multiple issues:

1. i discovered that `config.theme.timezone` is undefined, when running this test. That shouldn't happen at all, because it can cause writing wrong tests. I will push a fix for that soon.
2. we need to expect a created permalink based on your current blog TZ.
3. We can fix this without #6930, because we can assert a hardcoded permalink assertion to get the tests green.
